pkg/scm/gitlab/client_actioner.go

```diff
@@ -153,6 +155,81 @@ func (c *Client) ApplyStep(ctx context.Context, evalContext scm.EvalContext, upd
 
 		return err
 
+	case "assign_reviewers":
+		source, err := step.RequiredStringEnum("source", "codeowners")
+		if err != nil {
+			return err
+		}
+
+		desiredLimit, err := step.RequiredInt("limit")
+		if err != nil {
+			return err
+		}
+
+		mode, err := step.OptionalStringEnum("mode", "linear", "linear", "random")
+		if err != nil {
+			return err
+		}
+
+		var eligibleReviewers []scm.Actor
+
+		switch source {
+		case "codeowners":
+			eligibleReviewers = evalContext.GetCodeOwners()
+
+			break
+		}
+
+		if len(eligibleReviewers) == 0 {
+			return nil
+		}
+
+		var reviewers []scm.Actor
+
+		var limit int
+		if desiredLimit > len(eligibleReviewers) {
+			limit = len(eligibleReviewers)
+		} else {
+			limit = desiredLimit
+		}
+
+		switch mode {
+		case "linear":
+			reviewers = eligibleReviewers[:limit]
+
+			break
+		case "random":
+			reviewers = make([]scm.Actor, limit)
+			r := rand.New(rand.NewSource(time.Now().UnixNano()))
+			perm := r.Perm(len(eligibleReviewers))
+
+			for i := 0; i < limit; i++ {
+				reviewers[i] = eligibleReviewers[perm[i]]
+			}
+
+			break
+		}
+
+		var reviewerIDs []int
+
+		for _, reviewer := range reviewers {
+			reviewerIDs = append(reviewerIDs, reviewer.IntID())
+		}
+
+		if state.IsDryRun(ctx) {
+			slogctx.Info(ctx, "(Dry Run) Assigning MR", slog.String("source", source), slog.Int("limit", limit), slog.String("mode", mode), slog.Any("reviewers", reviewers))
+
+			return nil
+		}
+
+		update := &scm.UpdateMergeRequestOptions{
+			ReviewerIDs: &reviewerIDs,
+		}
+
+		_, err = c.MergeRequests().Update(ctx, update)
+
+		return err
```

Comment:
There is an `scm.UpdateMergeRequestOptions()` passed into the action handler, and all the actions (where possible) sets their properties - so when its submitted to GitLab, it kinda rolls up many actions into one API call bulk changing everything

One thing worth considering is whether we should append to `update.ReviewerIDs()` - so if you have to `action` logic blocks, they are handled properly - especially when more options for assigning reviewers are added - like a static list

```yaml
    then:
      - action: assign_reviewers
        source: codeowners
        limit: 1
        mode: random
      - action: assign_reviewers
        source: codeowners
        limit: 1
        mode: random
```


```suggestion
		update.ReviewerIDs = scm.Ptr(reviewerIDs)

		return nil
```

Reply:
updated and added an append reviewer ids method